### ISSUE TITLE
Mode keys stuck while animation is playing

### ISSUE THREAD

**Issue by niabot** [ckb-next/ckb-next#439]
## General information
OS: Ubuntu 18.04
Version: ckb-next 0.4.0
Source: http://ppa.launchpad.net/davidmeikle/ckb-next-daily/ubuntu
Keyboard: Corsair K70 RGB MK.2

## What is the program's current behaviour?
If i use animations, then the mode keys (Caps Lock, Num Lock, ...) won't change until the animation has finished.
For example: I applied the _Type lighting_ effect with a duration of 2 seconds to all keys. If i then press the X Key (the fading animation starts playing) and immediately press Caps Lock, then Caps Lock won't be active and won't change color until the animation is finished. Once the animation is finished, Caps Lock will be applied as many times as i pressed it during the animation duration.
This isn't as terrible, but once i use an infinitely looping animation, it won't apply Caps or Num Lock anymore, since the state will only change after the animation is finished.  Which won't happen in this case.

## What is the expected behaviour of the program?
The Mode keys should be updated while an animation is playing.

**Comment by tatokis** [comment]
Duplicate of https://github.com/ckb-next/ckb-next/issues/391
